NuRadioMC/SignalGen/emitter.py

```diff
@@ -35,30 +44,42 @@ def get_time_trace(amplitude, N, dt, model, full_output=False, **kwargs):
         * idl1 & hvsp1 : these are the waveforms generated in KU lab and stored in hdf5 files
         * gaussian : represents a gaussian pulse where sigma is defined through the half width at half maximum
         * ARA02-calPulser : a new normalized voltage signal which depicts the original CalPulser shape used in ARA-02
+        * efield_idl1_spice: direct measurement of the efield from the idl1 pulser and its antenna as used in the SPICE
+          calibration campaigns from 2018 and 2019. 
+          The `launch_vector` needs to be specified in the kwargs. See Journal of Instrumentation 15 (2020) P09039,
+          doi:10.1088/1748-0221/15/09/P09039 arXiv:2006.03027 for details.
+        * efield_delta_pulse: a simple signal model of a delta pulse emitter. The kwarg `polarization` needs
+          to be specified to select the polarization of the efield, defined as float between 0 and 1 with
+          0 = eTheta polarized and 1 = ePhi polarized. The default is 0.5, i.e. unpolarized. The amplitudes are
+          set to preserve the total power of the delta pulse, i.e. A_theta = sqrt(1-polarization)
+          and A_phi = sqrt(polarization).
+          Use kwarg `iN` to select a specific pulse from the 10 available pulses. The default is a random selection.
     full_output: bool (default False)
         if True, can return additional output
 
     Returns
     -------
-    time trace: 2d array, shape (3, N)
-        the amplitudes for each time bin
+    time trace: 1d or 2d array, shape (N) or (3, N) for efield
+        the amplitudes for each time bin. In case of an efield, the the amplitude for the three componente eR, eTheta, ePhi are returned.
     additional information: dict
         only available if `full_output` enabled
 
     """
-    half_width = kwargs.get("half_width")
-    emitter_frequency = kwargs.get("emitter_frequency")
     trace = None
     additional_output = {}
     if(amplitude == 0):
-        trace = np.zeros(3, N)
+        if(model.startswith("efield_")):
+            trace = np.zeros((3, N))
+        else:
+            trace = np.zeros(N)
     if(model == 'delta_pulse'):  # this takes delta signal as input voltage
         trace = np.zeros(N)
         trace[N // 2] = amplitude
     elif(model == 'cw'):  # generates a sine wave of given frequency
         time = np.linspace(-(N / 2) * dt, ((N - 1) - N / 2) * dt, N)
-        trace = amplitude * np.sin(2 * np.pi * emitter_frequency * time)
+        trace = amplitude * np.sin(2 * np.pi * kwargs["emitter_frequency"] * time)
     elif(model == 'square' or model == 'tone_burst'):  # generates a rectangular or tone_burst signal of given width and frequency
+        half_width = kwargs.get("half_width")
```

Comment:
Consider raising a more helpful error if a kwarg is expected but not specified. Also, document the kwargs in the docstring, either under Parameters or Other Parameters.

Reply:
fixed